python/cudf/cudf/core/series.py

```diff
@@ -4106,26 +4106,31 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
     dtype: bool
     """
 
+    if isinstance(a, (cudf.Series, pd.Series)) and isinstance(
+        b, (cudf.Series, pd.Series)
+    ):
+        b = b.reindex(a.index)
```

Comment:
I think we should just assert that both `a` and `b` are `cudf.Series` or always convert to a `cudf.Series`. We'd ideally want to do this zero copy if possible though.

If `b` is a Pandas series it calls Pandas's `reindex` here.

Reply:
Validating inputs to always be a `cudf.Series` but also put in a check to convert `pd.Series` to `cudf.Series`.

Reply:
This is another situation where we'd want to handle anything that we can nicely convert to a Series like cupy arrays or numpy arrays or Python lists, etc.

Reply:
The latest changes will now support all of them 👍